### ISSUE TITLE
Cannot edit docx document in Word Online

### ISSUE THREAD

**Issue by donmeuse** [WordDocX/DocX#103]
I tested 2 documents in Word Online, the first one was the one created in the DocX tutorial video, and second one created using the new DocXCore port version. I transferred them to One Drive and opened them in Word Online . I can see the text when Word Online first loads in "View" mode. But when I select "Edit in Word Online" the document is blank - no text at all.

MORE DETAILS: From within Word Online if I select "Edit in Word" it loads in my local Word 2010 properly and I can edit and save without issue, all text is visible.

If I edit the "blank" doc in Word Online, for example if I add a line "This is a test" to the "blank" document and save it, then open it in Word locally I see "This is a test" plus all the old text, that is still blank in Online. So the old text is there, just not visible in Edit mode.

I then created a doc in Word 2010 and transferred to One Drive and it loaded and edited normally in Word Online. No issues with that, only if I create it with DocX or DocXCore. I will install Office365 in the new few days to see how that reacts.

The "Version Requirements" for Word Online appear to be both .Doc and .DocX versions, there are no restrictions or exceptions I could find.

Comments or suggestions, anyone?

**Comment by donmeuse** [comment]
Follow-up: I installed Office 365 and tested the files created in DocX - they loaded from OneDrive and worked normally, I could edit them. If I saved changes from Office 365 I could then edit the files in Word Online. So there is something changed that changes the doc structure when saved in 365 vs. DocX.

**Comment by PrzemyslawKlys** [comment]
You can always take a file "resave it" in office 365 without changing things and compare what changed (just change .docx to .zip and unpack files - and then compare it with some tool). This will narrow it down for you. We don't know how Office Online works thou...

**Comment by donmeuse** [comment]
I loaded and resaved in Office 365 and there are additional files created that don't exist in the original DocX version. I'm new at this so this is beyond my skill set. Thanks for the advice!

**Comment by tseward** [comment]
Yes, this does appear to also be an issue in Office Online Server. While the files can be viewed in OOS, once entering edit mode, it fails to render the content. But this _appears_ to be due to the Compatibility Mode the document is in. Upgrading the document to the current rev allows the file to be edited within OOS.

**Comment by donmeuse** [comment]
Thanks, that is what I found originally.  I'm testing how to use Docx to create the documents programmatically on the server and load directly into Word Online for the user. In that scenario I have no ability to programmatically load in Office and save back as current Word version.

**Comment by rzwnk** [comment]
Please try to fix this issue as this is a huge issue even I am not able to edit my word document even though I have bought a office subsription just to do so.

**Comment by PrzemyslawKlys** [comment]
To be honest it does work for me. 

Word document created with PSWriteWord (https://github.com/EvotecIT/PSWriteWord) which utilizes this library, copied to onedrive and opened in Office 365 browser

![image](https://user-images.githubusercontent.com/15063294/43162437-4ac718f8-8f8b-11e8-940c-846f6003b4e6.png)

Then added text to it via Office 365 and opened on desktop

![image](https://user-images.githubusercontent.com/15063294/43162509-8c63c93c-8f8b-11e8-9bbb-2716de742f75.png)

So Word Online does work fine. Onedrive seems to work as well.

**Comment by zimmermanw84** [comment]
I've had a similar issue. When I open it in Word it says the file is corrupted, and the Office Online error below. My hack around this was to open the file (after Docx manipulation and saving) using `DocumentFormat.OpenXml.Wordprocessing` and save it fixed my issue. Seems like the Docx is saving it in another format (ODF?), but for whatever reason OpenXml can save to the correct format.

It's possible that this is an issue because I am running DotNet Core 2.1.

<img width="574" alt="auditboard___bp___as_in_1_c18" src="https://user-images.githubusercontent.com/8593005/49049288-47175c80-f193-11e8-8c6c-39051d88c80f.png">

**Comment by XceedBoucherS** [comment]
Xceed Words for .NET will only be available for .NET Core applications in v1.6.
